test/format/config.c

```diff
@@ -285,6 +285,23 @@ config_table(TABLE *table, void *arg)
             config_single(table, "btree.value_min=20", false);
     }
 
+    /*
+     * Limit the rows to one million if the realloc exact and realloc malloc configs are on and not
+     * all explicitly set. Realloc exact config allocates the exact amount of memory, which causes a
+     * new realloc call every time we append to an array. Realloc malloc turns a single realloc call
+     * to a malloc, a memcpy, and a free. The combination of both will significantly slow the
+     * execution.
+     */
+    if ((!config_explicit(NULL, "debug.realloc_exact") ||
+          !config_explicit(NULL, "debug.realloc_malloc")) &&
+      GV(DEBUG_REALLOC_EXACT) && GV(DEBUG_REALLOC_MALLOC) && TV(RUNS_ROWS) > WT_MILLION) {
+        config_single(table, "runs.rows=" STR(WT_MILLION), true);
+        WARN(
+          "limiting table%" PRIu32
+          ".runs.rows to one million if realloc_exact or realloc_malloc has been automatically set",
```

Comment:
Minor: Instead of 'one million' maybe have it `% PRIu32` for `WT_MILLION. That way if we ever change it to WT_MILLION * 2 or WT_MILLION/2 it is easy to find and replace.